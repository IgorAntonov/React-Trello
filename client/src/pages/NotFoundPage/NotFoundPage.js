import React from 'react';
import posed from 'react-pose';
import {
  Col,
  FlexPage,
  H1,
  H2,
  Footer,
  StyledLink
} from 'Src/ui';

const PosedH1 = posed(H1)({
  init: { x: '-300%' },
  mount: { x: 0, transition: { type: 'spring', delay: 100 } }
});
const PosedH2 = posed(H2)({
  init: { x: '100%' },
  mount: { x: 0, transition: { type: 'spring', delay: 100 } }
});

export const NotFoundPage = () => (
  <FlexPage>
    <Col grow={1} justify="center" align="center" bgColor="#89609e" >
      <PosedH1 initialPose="init" pose="mount">404</PosedH1>
      <PosedH2 initialPose="init" pose="mount" marginB="2rem" >Page not found :(</PosedH2>
      <StyledLink to="/" bgcolor="#755286" >Take me back to the main page</StyledLink>
    </Col>
    <Footer />
  </FlexPage>
);
