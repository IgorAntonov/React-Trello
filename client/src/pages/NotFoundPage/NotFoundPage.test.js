import React from 'react';
import { shallow } from 'enzyme';

import { NotFoundPage } from './NotFoundPage';

describe('<NotFoundPage/>', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<NotFoundPage />);
  });

  test('renders', () => {
    expect(wrapper.exists()).toBe(true);
  });
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
