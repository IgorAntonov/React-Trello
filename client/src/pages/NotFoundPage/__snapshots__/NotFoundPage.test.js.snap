// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage/> should match snapshot 1`] = `
<FlexPage>
  <styled-layout__Col
    align="center"
    bgColor="#89609e"
    grow={1}
    justify="center"
  >
    <Component
      initialPose="init"
      pose="mount"
    >
      404
    </Component>
    <Component
      initialPose="init"
      marginB="2rem"
      pose="mount"
    >
      Page not found :(
    </Component>
    <StyledLink
      bgcolor="#755286"
      to="/"
    >
      Take me back to the main page
    </StyledLink>
  </styled-layout__Col>
  <Footer />
</FlexPage>
`;
